### PR TITLE
Document that token-based upload is supported

### PR DESCRIPTION
**Pull request by Mariatta**
Closes https://github.com/pypa/flit/issues/276

### CODE REVIEW

**Review by pradyunsg — APPROVED** [review]
Thanks @Mariatta!

File: doc/cmdline.rst [hunk 1]
```diff
@@ -207,6 +207,10 @@ Environment variables
    Set a username, password, and index URL for uploading packages.
    See :ref:`uploading packages with environment variables <upload_envvars>`
    for more information.
+   
+   Token-based upload to PyPI is supported. To upload using PyPI token,
+   set the ``FLIT_USERNAME`` value to ``__token__``, and the ``FLIT_PASSWORD``
+   to the token value.
```
Comment (joukewitteveen) [inline doc/cmdline.rst:213]:
```suggestion
   Token-based upload to PyPI is supported. To upload using a PyPI token,
   set ``FLIT_USERNAME`` to ``__token__``, and ``FLIT_PASSWORD`` to the
   token value.
```

**Review by joukewitteveen — COMMENTED** [review]
Not sure if this needs to be in the flit documentation, since it is not specific to flit.

**Comment by Mariatta** [comment]
I was publishing using flit, and it was not obvious to me that I could use the PyPI token instead of regular username/password. Additionally, while searching for "upload using token with flit" I came across this open issue on GitHub which made me think that it was not supported. So I thought that it would be better to explicitly document it.

**Comment by pradyunsg** [comment]
To err on the side of overcommunicating, I'm unlikely to be merging this PR.

I have a green tick cos I'm an admin on pypa but I'm not a committer to twine, and would prefer that someone who is lands this instead. :)

**Comment by takluyver** [comment]
Thanks @Mariatta !

@pradyunsg FWIW, I'm happy for you to merge something like this if you're comfortable doing so.

**Pull request merged**